scheduler/scheduler-server/src/test/java/functionaltests/policy/license/TestLicensePolicyWithRestart.java

```diff
@@ -0,0 +1,104 @@
+ * and Big Data Analysis on Enterprise Grids & Clouds.
+ *
+ * Copyright (c) 2007 - 2017 ActiveEon
+ * Contact: contact@activeeon.com
+ *
+ * This library is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU Affero General Public License
+ * as published by the Free Software Foundation: version 3 of
+ * the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package functionaltests.policy.license;
+
+import static functionaltests.utils.SchedulerTHelper.log;
+
+import java.io.File;
+import java.net.URL;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.ow2.proactive.scheduler.common.Scheduler;
+import org.ow2.proactive.scheduler.common.job.JobId;
+import org.ow2.proactive.scheduler.common.job.JobState;
+import org.python.google.common.collect.ImmutableMap;
+
+import functionaltests.utils.SchedulerFunctionalTestLicensePolicy;
+import functionaltests.utils.SchedulerTHelper;
+
+
+public class TestLicensePolicyWithRestart extends SchedulerFunctionalTestLicensePolicy {
+
+    private static URL JobSimpleJobLicensePolicy = TestLicensePolicyWithRestart.class.getResource("/functionaltests/descriptors/Job_simple_job_license_policy.xml");
+
+    @Test
+    public void testRestoreTokensAfterSchedulerRestart() throws Throwable {
+
+        log("Starting testRestoreTokensAfterSchedulerRestart");
+
+        JobId jobId0 = schedulerHelper.submitJob(new File(JobSimpleJobLicensePolicy.toURI()).getAbsolutePath(),
+                                                 ImmutableMap.of("LICENSES", "software_A"));
+        JobId jobId1 = schedulerHelper.submitJob(new File(JobSimpleJobLicensePolicy.toURI()).getAbsolutePath(),
+                                                 ImmutableMap.of("LICENSES", "software_B"));
+
+        JobId jobId2 = schedulerHelper.submitJob(new File(JobSimpleJobLicensePolicy.toURI()).getAbsolutePath(),
+                                                 ImmutableMap.of("LICENSES", "software_A"));
+
+        JobId jobId3 = schedulerHelper.submitJob(new File(JobSimpleJobLicensePolicy.toURI()).getAbsolutePath(),
+                                                 ImmutableMap.of("LICENSES", "software_B"));
+
+        log("Waiting for jobs submitted");
+        schedulerHelper.waitForEventJobSubmitted(jobId0);
+        schedulerHelper.waitForEventJobSubmitted(jobId1);
+        schedulerHelper.waitForEventJobSubmitted(jobId2);
+        schedulerHelper.waitForEventJobSubmitted(jobId3);
```

Comment:
shouldn't you wait for the first job to be started before restarting the scheduler ?